R/pkg/R/SQLContext.R

```diff
@@ -147,6 +147,55 @@ getDefaultSqlSource <- function() {
   l[["spark.sql.sources.default"]]
 }
 
+writeToTempFileInArrow <- function(rdf, numPartitions) {
+  # R API in Arrow is not yet released. CRAN requires to add the package in requireNamespace
+  # at DESCRIPTION. Later, CRAN checks if the package is available or not. Therefore, it works
+  # around by avoiding direct requireNamespace.
+  requireNamespace1 <- requireNamespace
+  if (requireNamespace1("arrow", quietly = TRUE)) {
+    record_batch <- get("record_batch", envir = asNamespace("arrow"), inherits = FALSE)
+    record_batch_stream_writer <- get(
+      "record_batch_stream_writer", envir = asNamespace("arrow"), inherits = FALSE)
+    file_output_stream <- get(
+      "file_output_stream", envir = asNamespace("arrow"), inherits = FALSE)
+    write_record_batch <- get(
+      "write_record_batch", envir = asNamespace("arrow"), inherits = FALSE)
+
+    # Currently arrow requires withr; otherwise, write APIs don't work.
+    # Direct 'require' is not recommended by CRAN. Here's a workaround.
+    require1 <- require
+    if (require1("withr", quietly = TRUE)) {
```

Comment:
It's actually a bit odd that I need to manually require this package. Otherwise, it complains, for instance, here https://github.com/apache/arrow/blob/d3ec69069649013229366ebe01e22f389597dc19/r/R/on_exit.R#L20

Reply:
require1 sounds a bit like a hack though...

Reply:
Yup .. it is .. looks we shouldn't have this error from a cursory look in R API of Arrow. Maybe this can be gone when I use official R Arrow release version. Let me check it later.

Reply:
that will be good. circumventing CRAN check for method name is... problematic..
(there are other more hacky way too, but ..)